packages/node/src/loader.ts

```diff
@@ -0,0 +1,67 @@
+import { defaultStylableMatcher } from './common';
+import nodeFs from '@file-services/node';
+import { stylableModuleFactory } from '@stylable/module-utils';
+import { loadStylableConfigEsm } from '@stylable/build-tools';
+import { resolveNamespace } from './resolve-namespace';
+import { fileURLToPath } from 'url';
+
+let createModule: ReturnType<typeof stylableModuleFactory>;
+async function generateJsModule(sheetSource: string, filePath: string) {
+    if (!createModule) {
+        createModule = await initiateModuleFactory();
+    }
+    return createModule(sheetSource, filePath);
+}
+async function initiateModuleFactory() {
+    const defaultConfig = await loadStylableConfigEsm(process.cwd(), (potentialConfigModule: any) =>
+        potentialConfigModule.defaultConfig?.(nodeFs)
+    );
+    return stylableModuleFactory(
```

Comment:
Where is the depth?

Reply:
Depth is not related to the loader. The factory can now use the CSS calculated depth that is available on the meta, but I would prefer to separate this into a different PR.